components/tidb_query_common/src/storage/scanner.rs

```diff
@@ -65,6 +65,15 @@ impl<T: Storage> RangesScanner<T> {
     // Note: This is not implemented over `Iterator` since it can fail.
     // TODO: Change to use reference to avoid allocation and copy.
     pub fn next(&mut self) -> Result<Option<OwnedKvPair>, StorageError> {
+        self.next_row(true)
+    }
+
+    /// Fetches next row.
+    /// Note: `update_scanned_range` can control whether update the scanned range when `is_scanned_range_aware` is true.
+    pub fn next_row(
```

Comment:
The name difference between `next` and `next_row` may be confusing. I don't have a good idea, but at least `next_opt` is better?